### PR TITLE
Fix parallel build

### PR DESCRIPTION
**Pull request by xiaoxiang781216**

### CODE REVIEW

**Comment by masayuki2009** [comment]
@xiaoxiang781216 I did parallel build aging test (from -j1 to -j24) for maix-bit:elf with this PR, but it failed.

```
lf_main.c: In function 'elf_main':                                                                                                                                                     
elf_main.c:270:41: error: 'romfs_img' undeclared (first use in this function)                                                                                                           
                          (FAR uint8_t *)romfs_img,                                                                                                                                     
                                         ^~~~~~~~~                                                                                                                                      
elf_main.c:270:41: note: each undeclared identifier is reported only once for each function it appears in                                                                               
elf_main.c:271:35: error: 'romfs_img_len' undeclared (first use in this function)                                                                                                       
                          NSECTORS(romfs_img_len), SECTORSIZE);                                                                                                                         
                                   ^~~~~~~~~~~~~                                                                                                                                        
elf_main.c:94:27: note: in definition of macro 'NSECTORS'                                                                                                                               
 #  define NSECTORS(b)  (((b) + SECTORSIZE - 1) / SECTORSIZE)          
```

**Comment by masayuki2009** [comment]
@xiaoxiang781216 Also, sometimes following error occurred which is the same as I commented in https://github.com/apache/incubator-nuttx-apps/pull/118#discussion_r392831232

```
riscv64-unknown-elf-ld: task.o: file not recognized: file truncated                                                                                                                     
Makefile:82: recipe for target 'task' failed                                                                                                                                            
make[4]: *** [task] Error 1                                                                                                                                               
```

**Comment by xiaoxiang781216** [comment]
@masayuki2009 how about the new change?

**Comment by masayuki2009** [comment]
Hi, @xiaoxiang781216 

> @masayuki2009 how about the new change?

So far, parallel build aging test for maix-bit:posix_spawn has no problem (FYI, there were no build errors for maix-bit:posix_spawn before applying the commit with my machine). 

However, maix-bit:elf still has build errors, because the commit does not affect examples/elf.

**Comment by xiaoxiang781216** [comment]
> Hi, @xiaoxiang781216
> 
> > @masayuki2009 how about the new change?
> 
> So far, parallel build aging test for maix-bit:posix_spawn has no problem (FYI, there were no build errors for maix-bit:posix_spawn before applying the commit with my machine).
> 
> However, maix-bit:elf still has build errors, because the commit does not affect examples/elf.

But the 2nd patch modify examples/elf/Makefile:
image

**Comment by masayuki2009** [comment]
@xiaoxiang781216 

> But the 2nd patch modify examples/elf/Makefile:

Sorry, I misunderstood. You are right.

> Do you cherry-pick the first two patch in PR?

I checked out your repository with the branch and confirmed the all commit ids are correct.
However, there still have errors for maixbit:elf

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216
> 
> > But the 2nd patch modify examples/elf/Makefile:
> 
> Sorry, I misunderstood. You are right.
> 
> > Do you cherry-pick the first two patch in PR?
> 
> I checked out your repository with the branch and confirmed the all commit ids are correct.
> However, there still have errors for maixbit:elf

Is it g_elf_exports not found? or something else?

**Comment by masayuki2009** [comment]
@xiaoxiang781216 

> Is it g_elf_exports not found? or something else?

```
riscv64-unknown-elf-ld: pthread.o: file not recognized: file truncated                                                                                                     
Makefile:82: recipe for target 'pthread' failed                                                                                                                            
make[4]: *** [pthread] Error 1               
```

**Comment by masayuki2009** [comment]
@xiaoxiang781216 

And another error occurred.

```
install task /mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests/romfs/task                                                                                         
install: cannot create regular file '/mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests/romfs/task': File exists                                                   
Makefile:90: recipe for target 'install' failed                                                                                                                            
```

File: examples/elf/tests/Makefile [hunk 1]
```diff
@@ -100,19 +100,10 @@ ifneq ($(CONFIG_EXAMPLES_ELF_NOSTRIP),y)
...
 endif
 
-# Create the file system image directory
-
-$(FSIMG_DIR):
-	$(Q) mkdir $(FSIMG_DIR)
-
-# Populate the file system image directory
-
-populate:  $(FSIMG_DIR) build install
```
Comment (xiaoxiang781216) [inline examples/elf/tests/Makefile]: @masayuki2009 I think I found the root cause why object files get truncated:
The populate target above trigger build and install run parallelly, and then gcc may compile the same source file twice concurrently.
Please try my new PR, which modify three similar examples(elf/posix_spwan/sotest).

Comment (masayuki2009) [reply examples/elf/tests/Makefile]: @xiaoxiang781216 I tried the new PR, but it failed.

```
riscv64-unknown-elf-ld: task.o: file not recognized: file truncated                                                                                                                                    
Makefile:82: recipe for target 'task' failed                                                                                                                                                           
make[4]: *** [task] Error 1                                                                                                                                                                            
make[4]: Leaving directory '/mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests/task'                                                                                                           
Makefile:88: recipe for target 'task_install' failed                                                                                                                                                   
make[3]: *** [task_install] Error 2                                                                                                                                                                    
make[3]: Leaving directory '/mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests'                                                                                                                
Makefile:57: recipe for target 'tests/symtab.c' failed                                              
```

**Pull request closed without merging**